lib/tools.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+/*
+ * A bunch of handy functions that should probably live in a module somewhere.
+ */
+
+function filterByKey(object, filter) {
+    var filtered = {};
+    Object.keys(object).forEach(function (key) {
+        if (filter(key)) {
+            filtered[key] = object[key];
+        }
+    });
+    return filtered;
+}
+
+function find(xs, predicate) {
```

Comment:
Thanks for pointing that out - I have much to learn about the various JS std libs.  I'll switch to that and toss out this ad-hoc implementation.